safe-global/safe-wallet-web
# fix: Set Safe apps categories via URL query

## What it solves

Resolves #

This is an enhancement for the safe apps list and allows to set categories via the URL. It will be for linking to certain categories like Stake, Bridge, Swap e.g. from the Dashboard.

## How this PR fixes it

- Adds a new hook `useCategoryFilter` that checks the URL for `categories` and syncs it with the local state

## ToDo

- [x] Write tests for `useCategoryFilter`

## How to test it

1. Create a counterfactual Safe
2. Press Create Transaction on the dashboard
3. Select Swap tokens
4. Observe being navigated to the safe apps list with `DEX` apps pre-selected
5. Clear the category
6. Observe the URL and list updating

## Screenshots
<img width="1273" alt="Screenshot 2024-02-16 at 15 58 41" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/769ff83a-819f-4dd1-abb0-37914f11914d">

## Checklist
